### PR TITLE
feat(protocol-designer): Add thermocycler state form

### LINKED ISSUE #5568: TC Step Form (State Fields)
As a PD user, I would like to be able to create a basic state change step for the Thermocycler

## Acceptance Criteria
- [ ] User can toggle the block between “Active” and “Deactivated”
  - [ ] When toggled to “Active”, user can see and edit a temperature field to set TC block temp
- [ ] User can toggle the lid between “Active” and “Deactivated”
  - [ ] When toggled to “Active”, user can see and edit a temperature field to set TC block temp
- [ ] User can toggle the lid between “Open” and “Closed”
- [ ] User can click “Close” to close the form without saving data

## Design
- [All fields deactivated (Invision)](https://projects.invisionapp.com/share/VUWZT27EDJ8#/screens/414817325)
- [Lid and block fields completed (Invision)](https://projects.invisionapp.com/share/VUWZT27EDJ8#/screens/414817327)

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR closes #5568 by adding the form fields for setting TC state.

## changelog

- feat(protocol-designer): Add thermocycler state form

## review requests

Make a protocol with a TC
Add a TC step
Select change thermocycler state
- [ ] State fields conditionally render
- [ ] Labels are accurate
- [ ] Temperature fields conditionally render when lid or block is "active"

## risk assessment

Low, just a form in PD that doesn't do anything yet + behind a FF

### CODE REVIEW

File: protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// @flow
+import * as React from 'react'
+import { ToggleField } from '@opentrons/components'
+import cx from 'classnames'
+
+import type { StepFieldName } from '../../../steplist/fieldLevel'
+import styles from '../StepEditForm.css'
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js:7]: type should go after `FieldConnector ` as per https://github.com/Opentrons/opentrons/pull/4680#discussion_r363102698

We really need a linter for this stuff... I'll make a ticket

File: protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { FieldConnector } from './FieldConnector'
+
+type ToggleRowProps = {
+  labelOff?: string,
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js:11]: exact type

File: protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  labelOn?: string,
```
Comment (shlokamin) [inline protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js:12]: can we rename `labelOn` and `labelOff` to `onLabel` and `offLabel` or something to try to indicate they're not booleans?

**Review by shlokamin — CHANGES_REQUESTED** [review]
Small nitpicks. Also, should we add test coverage around `Thermocycler Form` stuff like we do for `MagnetForm`?

File: protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js (hunk 1 above)
Comment (shlokamin) [reply protocol-designer/src/components/StepEditForm/fields/ToggleRowField.js:7]: https://github.com/Opentrons/opentrons/issues/5624

**Review by IanLondon — COMMENTED** [review]
looks good, I agree w/ Shlok's comments

**Review by shlokamin — APPROVED** [review]
Toggle component looks snazzy! Bump on my comment about test coverage tho, I personally think we should unit test form components going forward.

**Comment by Kadee80** [comment]
> Toggle component looks snazzy! Bump on my comment about test coverage tho, I personally think we should unit test form components going forward.

Totally agree, was holding off on testing till we had validation in place

**Review by IanLondon — APPROVED** [review]
:+1:

**Pull request merged**